### PR TITLE
bt-211: add no connection to the server component

### LINKED ISSUE #211: Add no connection to the server component
We need to add a check if the user has an internet connection or connection to the server and if not, display that info in a popup (which can be placed above screen headers). Research ways how we can do it and create different UI variants for popup and discuss them with the team

### PR DESCRIPTION
**Pull request by mskmee**

### CODE REVIEW

File: mobile/src/app/app.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import SplashScreen from 'react-native-splash-screen';
 import Toast from 'react-native-toast-message';
 import { Provider as StoreProvider } from 'react-redux';
 
+import { NetworkDisableModal } from '~/bundles/common/components/components';
```
Comment (katerynakostikova) [inline mobile/src/app/app.tsx]: LostConnectionModal

File: mobile/src/bundles/common/components/network-dissable-modal/network-dissable-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import NetInfo from '@react-native-community/netinfo';
+import React from 'react';
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
+
+import { Text, View } from '~/bundles/common/components/components';
+import { IconName, TextCategory } from '~/bundles/common/enums/enums';
+import { useEffect, useState } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/network-dissable-modal/network-dissable-modal.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles;
```

File: mobile/src/bundles/common/components/network-dissable-modal/network-dissable-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            const isOnline = !(state.isConnected && state.isInternetReachable);
+            setIsConnected(isOnline);
+        });
+        return () => {
+            unsubscribe();
+        };
+    }, []);
+    if (!isConnected) {
+        return null;
+    }
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/network-dissable-modal/network-dissable-modal.tsx]: Can we use `useNetInfo()` ?

Comment (mskmee) [reply mobile/src/bundles/common/components/network-dissable-modal/network-dissable-modal.tsx]: yeap, thanks for advice)

File: mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+import { Text, View } from '~/bundles/common/components/components';
+import { IconName, TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/styles';
+
+import { styles } from './styles';
+
+const LostConnectionModal: React.FC = () => {
+    const { isConnected, isInternetReachable } = useNetInfo();
+    const isConnectionLos = !isConnected && !isInternetReachable;
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx]:
```suggestion
    const isConnectionLost = !isConnected && !isInternetReachable;
```

File: mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                    globalStyles.m25,
+                    globalStyles.p10,
+                    globalStyles.borderRadius10,
+                    globalStyles.justifyContentSpaceAround,
+                    globalStyles.alignItemsCenter,
+                ]}
+            >
+                <Icon name={IconName.WIFI_OFF} size={25} color={'#fff'} />
+                <Text style={styles.message} category={TextCategory.MENU}>
+                    Internet connection error
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx]:
```suggestion
                    No connection to the server
```

File: mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    if (!isConnectionLos) {
+        return null;
+    }
+
+    return (
+        <View
+            pointerEvents="box-only"
+            style={[
+                globalStyles.width100,
+                globalStyles.height100,
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx:24]: why 100 here?

Comment (mskmee) [reply mobile/src/bundles/common/components/lost-connection-modal/lost-connection-modal.tsx:24]: It's an absolute element, which didn't let user change anything

**Review by katerynakostikova — APPROVED** [review]

**Comment by katerynakostikova** [comment]
Let's just improve UI a bit
Let's position items in the center and add some right margin to the icon. There is a big gap between icon and text

**Comment by mskmee** [comment]
> Let's just improve UI a bit Let's position items in the center and add some right margin to the icon. There is a big gap between icon and text

image
Refactor styles

**Comment by katerynakostikova** [comment]
Let's leave it on the top of the screen

**Review by anogire — APPROVED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

**Review by nikita-remeslov — APPROVED** [review]

**Comment by mskmee** [comment]
Return it back to top 
image

**Pull request merged**